### PR TITLE
Display the version of runner in the runner list

### LINKED ISSUE #23489: Display the version of runner in runner list page.
### Feature Description

According to this [PR](https://gitea.com/gitea/act_runner/pulls/43), We can get the runner version from each request from the client. 
we can keep getting the latest version of the runner by displaying it in the runner list.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by sillyguodong**
Close: #23489 

### Change
1. Add version column to action_runner table.
2. Read the runner version from the request header, and update it in DB.
3. Display version in runner list

### Screenshot
image

### CODE REVIEW

File: routers/api/actions/runner/interceptor.go [hunk 1]
```diff
@@ -33,6 +36,10 @@ var withRunner = connect.WithInterceptors(connect.UnaryInterceptorFunc(func(unar
 		}
 		uuid := request.Header().Get(uuidHeaderKey)
 		token := request.Header().Get(tokenHeaderKey)
+		version := request.Header().Get(versionHeaderKey)
```
Comment (wolfogre) [inline routers/api/actions/runner/interceptor.go:39]: Should truncate the version if it's too long.

File: routers/api/actions/runner/interceptor.go [hunk 2]
```diff
@@ -44,7 +51,8 @@ var withRunner = connect.WithInterceptors(connect.UnaryInterceptorFunc(func(unar
 			return nil, status.Error(codes.Unauthenticated, "unregistered runner")
 		}
 
-		cols := []string{"last_online"}
+		cols := []string{"version", "last_online"}
+		runner.Version = version
```
Comment (wolfogre) [inline routers/api/actions/runner/interceptor.go]: It would be better to update version only if `runner.Version != version`

**Review by lunny — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Comment by delvh** [comment]
Please fix the conflicts so that this PR can be merged.

**Comment by sillyguodong** [comment]
> Please fix the conflicts so that this PR can be merged.

done

**Pull request merged**